simplematrixbotlib/match.py

```diff
@@ -156,11 +156,11 @@ def mention(self):
             Returns True if the message begins with the bot's username, MXID, or pill targeting the MXID, and False otherwise.
         """
 
-        for i in [self._display_name, self._disambiguated_name, self.room.own_user_id, self._pill]:
-            body = self.event.formatted_body or self.event.body
-            if body.startswith(i):
-                return True
-
+        for body in (self.event.formatted_body, self.event.body):
+            for id in [self._display_name, self._disambiguated_name, self.room.own_user_id, self._pill]:
+                if body.startswith(id):
+                    return True
+                    
```

Comment:
This is not a bug, here is what `body = self.event.formatted_body or self.event.body` does:
1. if `self.event.formatted_body` has a value, then it is assigned to `body`
2. if `self.event.formatted_body` is `None` (https://matrix-nio.readthedocs.io/en/latest/nio.html#module-nio.events.room_events) then continue to `or`
3. assign `body = self.event.body`

![image](https://user-images.githubusercontent.com/2803622/143427579-578bb155-2271-4294-bb28-8177b1f80ea3.png)

that is what I intended. right now the body can be `None` and you will into an exception with that.

however, if you argue that's bad style then that's something else I don't know so much about. clearly it wasn't readable to you, so there is a point.

something that I did miss and we should take away from here is to always check body still even if formatted body exists. i am thinking:
```
if body is None:
    continue
```

Reply:
I was not aware of how that worked, but I understand now. In Python, or only evaluates the second expression if the first expression does not return a "non none" value. That said, it was unclear to me, and thus likely to others as well. That specific part should be restored but implemented in a different way. Perhaps an explicit if/else ?

Reply:
I redid this entirely, please comment anew if you see any other issue with it